src/main/java/edu/indiana/dlib/amppd/repository/WorkflowResultRepositoryCustomImpl.java

```diff
@@ -215,20 +219,20 @@ private WorkflowResultFilterValues getFilterValues() {
         Root<WorkflowResult> rootDateCriteria = queryDate.from(WorkflowResult.class);
         CriteriaQuery<GalaxyJobState> queryGjs = cb.createQuery(GalaxyJobState.class);
         Root<WorkflowResult> rootGjs = queryGjs.from(WorkflowResult.class);
-        
+
         // We treat each filter independently, i.e. its possible value set is not dependent on current selected values in other filters;
         // rather, we populate each filter with distinct values existing in the current Workflow table.
         // Making filter value set context-dependent will result in deadlock queries. 
 
         List<Date> dateFilters = em.createQuery(queryDate.select(rootDateCriteria.get(DATE_PROPERTY).as(java.sql.Date.class))).getResultList();
         List<String> submitters = em.createQuery(query.select(root.get("submitter")).distinct(true)).getResultList();
-        List<String> collections = em.createQuery(query.select(root.get("collectionName")).distinct(true)).getResultList();
+        List<Object> collections = em.createQuery("FROM WorkflowResult WHERE id IN (SELECT MIN(id) FROM WorkflowResult GROUP BY collectionName)").getResultList();
         List<String> externalIds = em.createQuery(query.select(root.get("externalId")).distinct(true)).getResultList();
         List<String> items = em.createQuery(query.select(root.get("itemName")).distinct(true)).getResultList();
         List<String> files = em.createQuery(query.select(root.get("primaryfileName")).distinct(true)).getResultList();
-        List<String> workflows = em.createQuery(query.select(root.get("workflowName")).distinct(true)).getResultList();
+        List<Object> workflows = em.createQuery("FROM WorkflowResult WHERE id IN (SELECT MIN(id) FROM WorkflowResult GROUP BY workflowName)").getResultList();
         List<String> steps = em.createQuery(query.select(root.get("workflowStep")).distinct(true)).getResultList();
-        List<String> outputs = em.createQuery(query.select(root.get("outputName")).distinct(true)).getResultList();
+        List<Object> outputs = em.createQuery("FROM WorkflowResult WHERE id IN (SELECT MIN(id) FROM WorkflowResult GROUP BY outputName)").getResultList();
         List<GalaxyJobState> statuses = em.createQuery(queryGjs.select(rootGjs.get("status")).distinct(true)).getResultList();
```

Comment:
Hi Sarath, I know we reviewed this part online, but it was kind in a hurry and I didn't have time to think this through. Today I'm working on some other changes for Dashboard queries, and I reviewed this part again. I must say I really dislike using SQLs here; also, why do you need to change the logic here? Please discuss with me again, and I'd like you to change this part to not use SQLs, but use same pattern as other filters as used to be.